libs/ett/core/shell/src/lib/ett-shell.component.ts

```diff
@@ -35,27 +47,129 @@ const selector = 'ett-shell';
     `
       ${selector} {
         display: block;
+
+        .toolbar {
+          width: 100%;
+          display: flex;
+
+          h1 {
+            flex: auto;
+          }
+
+          .menu {
+            align-self: flex-end;
+          }
+        }
       }
     `,
   ],
   template: `
-    <watt-shell>
+    <div *ngIf="profileLoading">Loading</div>
+
+    <watt-shell *ngIf="!profileLoading">
       <ng-container watt-shell-sidenav>
         <ett-primary-navigation></ett-primary-navigation>
       </ng-container>
 
       <ng-container watt-shell-toolbar>
-        <h1>Energy Track and Trace</h1>
+        <div class="toolbar">
+          <h1>Energy Origin</h1>
+          <div class="menu">
+            <watt-button
+              type="text"
+              size="normal"
+              [disabled]="false"
+              [loading]="false"
+              [matMenuTriggerFor]="menu"
+            >
+              {{ profile?.name }}
+              <mat-icon aria-hidden="false" aria-label="Logout"
+                >expand_more</mat-icon
+              >
+            </watt-button>
+            <mat-menu #menu="matMenu">
+              <button mat-menu-item (click)="logout()">Logout</button>
+            </mat-menu>
+          </div>
+        </div>
       </ng-container>
 
       <router-outlet></router-outlet>
     </watt-shell>
   `,
 })
-export class EttShellComponent {}
+export class EttShellComponent implements OnInit {
+  profile?: UserProfile;
+  profileLoading = true;
+
+  constructor(
+    private router: Router,
+    private change: ChangeDetectorRef,
+    private authOidc: AuthOidcHttp
+  ) {}
+
+  ngOnInit() {
+    this.getProfile();
+  }
+
+  // -- Logout ---------------------------------------------------------------
+
+  logout() {
+    this.authOidc
+      .logout()
+      .pipe(map((response) => response.success))
+      .subscribe({
+        next: this.onLogoutComplete.bind(this),
+        error: this.onLogoutFailed.bind(this),
+      });
+  }
+
+  private onLogoutComplete(success: boolean) {
+    if (success) {
+      this.router.navigate(['/login']);
+    }
+  }
+
+  private onLogoutFailed(error: any) {
+    // TODO Test this
+    console.log('onLogoutFailed', error);
+  }
+
+  // -- GetProfile -----------------------------------------------------------
+
+  getProfile() {
+    this.profileLoading = true;
+    this.authOidc.getProfile().subscribe({
+      next: this.onGetProfileComplete.bind(this),
+      error: this.onGetProfileFailed.bind(this),
+    });
+  }
+
+  private onGetProfileComplete(response?: GetProfileResponse) {
+    if (response?.success) {
+      this.profile = response.profile;
+      this.profileLoading = false;
+      // TODO Replace with observables:
+      this.change.markForCheck();
+    }
+  }
+
+  private onGetProfileFailed(error: any) {
```

Comment:
issue: very rarely should we use the any type. Use unknown and figure out the actual type as needed through assertions or conditionals.